homeassistant/config_entries.py

```diff
@@ -1172,6 +1178,27 @@ async def _async_process_on_unload(self, hass: HomeAssistant) -> None:
                 task,
             )
 
+    @callback
+    def async_on_state_change(self, func: Callable[[], Any | None]) -> None:
+        """Add a function to call when a config entry changes its state."""
+        if self._on_state_change is None:
+            self._on_state_change = []
+        self._on_state_change.append(func)
+
+    def _async_process_on_state_change(self) -> None:
+        """Process the on_state_change callbacks and wait for pending tasks."""
+        if self._on_state_change is not None:
+            for func in self._on_state_change:
+                if func is not None:
```

Comment:
Its probably overkill to worry about `async_on_state_change` being called with `None`